src/main/java/Contexts/ContextHelper.java

```diff
@@ -0,0 +1,171 @@
+
+        return completed.trim();
+
+    }
+
+
+    public static ArrayList<String> getAllHints(String command , Controller controller) {
+        String [] splitCommand = command.toLowerCase().split(" ");
+        String incompleteCommand = getIncompleteWords(command.toLowerCase() , controller);
+
+        if (splitCommand.length == 0) {
+            return CommandContext.getRoot();
+        } else if (splitCommand.length == 1 && isRootCommandComplete(splitCommand[0])) {
+            ArrayList<String> allPossibilities =  CommandContext.getPossibilitiesSubRootForRoot(splitCommand[0]);
+            String update = completeCommand(allPossibilities , "");
+            ((MovieHandler) controller).updateTextField(update);
+            return allPossibilities;
+        } else if (splitCommand.length == 1) {
+            ArrayList<String> allPossibilities =  CommandContext.getPossibilitiesRoot(incompleteCommand);
+            String update = completeCommand(allPossibilities , incompleteCommand);
+            ((MovieHandler) controller).updateTextField(update);
+            return allPossibilities;
+        } else if (splitCommand.length == 2 && isSubRootCommandComplete(splitCommand[1])) {
+            ArrayList<String> hints  = commandSpecificHints(splitCommand[1]);
+//            ((MovieHandler)controller).updateTextField(update);
+
+            //TODO NOT COMPELTE and ROBUST
+            return hints;
+        } else if (splitCommand.length == 2) {
+            System.out.println("I TO FIND THE WORLD");
+            ArrayList<String> allPossibilities =  CommandContext.getPossibilitiesSubRoot(incompleteCommand);
+            String update = completeCommand(allPossibilities , incompleteCommand);
+            ((MovieHandler) controller).updateTextField(update);
+            return allPossibilities;
+        } else {
+            ArrayList<String> allPossibilities  = commandSpecificHints(splitCommand[0], splitCommand[1] , incompleteCommand);
+
+            String update = completeCommand(allPossibilities , incompleteCommand);
+            ((MovieHandler) controller).updateTextField(update);
+            return allPossibilities;
+        }
+
+    }
+
+    private static boolean isRootCommandComplete(String root) {
+        for (COMMANDKEYS c : CommandStructure.AllRoots) {
+            if (c.toString().equals(root)) {
+                return true;
+            }
+        }
+        return false;
+
+    }
+
+
+    private static boolean isSubRootCommandComplete(String subRoot) {
+        for (Map.Entry<COMMANDKEYS, COMMANDKEYS[]> e: CommandStructure.cmdStructure.entrySet()) {
+            for (COMMANDKEYS a: e.getValue()) {
+                if (a.toString().equals(subRoot)) {
+                    return true;
+                }
+            }
+
+        }
+        return false;
+
+    }
+
+    private static ArrayList<String> commandSpecificHints(String subRoot) {
+
+        switch(subRoot.toLowerCase().trim()) {
```

Comment:
What is this `subRoot.toLowerCase().trim()` for? Declare a variable for this so that readers know what's this about.